modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskTriggerHandler.java

```diff
@@ -246,7 +252,9 @@ private void unscheduleTaskRetry(String jobSubject) {
     private void handleError(Map<String, Object> params, Task task, TaskHandlerException e) {
         LOGGER.warn("Omitted task: {} with ID: {} because: {}", task.getName(), task.getId(), e);
 
-        activityService.addError(task, e, params);
+        if (!getTaskLogActivities().equals(LOG_NONE_ACTIVITIES)) {
```

Comment:
we should log error if a user want to log all or only error, so this condition is not easy to understand at a glance. More readable will be using methods which return boolean if we should log this activity